packages/meso-js/src/frame.ts

```diff
@@ -17,20 +23,61 @@ export const setupFrame = (
   };
 };
 
-const injectFullScreenIframe = (src: string) => {
+// Render the iframe for the "embedded" flow (full viewport modal)
+const renderIframe = (src: string, containerElement: Element | null) => {
   const iframe = document.createElement("iframe");
   iframe.src = src;
   iframe.setAttribute("allowtransparency", "true");
-  iframe.style.position = "fixed";
-  iframe.style.left = "0";
-  iframe.style.top = "0";
+
+  iframe.style.border = "none";
+  iframe.style.boxSizing = "border-box";
+  iframe.style.backgroundColor = "transparent";
+  iframe.style.colorScheme = "auto";
   iframe.style.width = "100%";
   iframe.style.height = "100%";
-  iframe.style.zIndex = "9999";
+
+  if (containerElement) {
+    containerElement.appendChild(iframe);
+  } else {
+    iframe.style.position = "fixed";
+    iframe.style.left = "0";
+    iframe.style.top = "0";
+    iframe.style.zIndex = "9999";
+
+    document.body.appendChild(iframe);
+  }
+
+  return iframe;
+};
+
+// Render an iframe for Onboarding in a modal experience
+export const renderModalOnboardingFrame = ({
+  apiHost,
+  pathname,
+}: {
+  apiHost: string;
+  pathname: string;
+}) => {
+  const route = pathname.startsWith(MODAL_ONBOARDING_PATH_PREFIX)
+    ? pathname
+    : `${MODAL_ONBOARDING_PATH_PREFIX}/${pathname}`;
+
+  const src = `${apiHost}${route.replaceAll("//", "/")}`;
+
+  const iframe = document.createElement("iframe");
+  iframe.src = src;
+  iframe.setAttribute("allowtransparency", "true");
+
   iframe.style.border = "none";
   iframe.style.boxSizing = "border-box";
   iframe.style.backgroundColor = "transparent";
   iframe.style.colorScheme = "auto";
+  iframe.style.width = "100%";
+  iframe.style.height = "100%";
+  iframe.style.position = "fixed";
+  iframe.style.left = "0";
+  iframe.style.top = "0";
+  iframe.style.zIndex = "9999";
```

Comment:
Nit: can we extract out a const for shared iframe styles?

Reply:
Updated